src/playdate/build/nimble.nim

```diff
@@ -0,0 +1,121 @@
+    if path.strip == "" or not path.strip.dirExists:
+        raise BuildFail.newException("Playdate nimble module is not a directory: " & path)
+    return path
+
+proc pdxName(): string =
+    ## The name of the pdx file to generate
+    projectName() & ".pdx"
+
+const SDK_ENV_VAR = "PLAYDATE_SDK_PATH"
+
+proc sdkPath(): string =
+    ## Returns the path of the playdate SDK
+    let fromEnv = getEnv(SDK_ENV_VAR)
+    let sdkPathCache = getConfigDir() / projectName() / SDK_ENV_VAR
+
+    if fromEnv != "":
+        mkDir(sdkPathCache.parentDir)
+        writeFile(sdkPathCache, fromEnv)
+        return fromEnv
+
+    if fileExists(sdkPathCache):
+        let fromFile = readFile(sdkPathCache)
+        if fromFile != "":
+            echo "Read SDK path from file: " & sdkPathCache
+            echo "SDK Path: " & fromFile
+            return fromFile
+
+    raise BuildFail.newException("SDK environment variable is not set: " & SDK_ENV_VAR)
+
+proc make(target: string) =
+    ## Executes a make target
+
+    let makefile = playdatePath() & "/playdate.mk"
+    if not makefile.fileExists:
+        raise BuildFail.newException("Could not find playdate Makefile: " & makefile)
+
+    putEnv(SDK_ENV_VAR, sdkPath())
+    putEnv("NIM_CACHE_DIR", nimcacheDir())
+    putEnv("PRODUCT", pdxName())
+    putEnv("UINCDIR", getCurrentCompilerExe().parentDir.parentDir / "lib")
+
+    let arch = if defined(macosx): "arch -arm64 " else: ""
+    exec(arch & "make " & target & " -f " & makefile)
+
+task cdevice, "build project":
+    nimble "-d:playdate", "build"
+
+task csim, "build project":
+    nimble "-d:simulator", "build"
+
+task simulator, "build project":
+    nimble "clean"
+    nimble "-d:simulator", "build"
+    make "pdc"
+
+task simulate, "build project the project and run the simulator":
+    nimble "simulator"
+    exec( (sdkPath() / "bin" / "PlaydateSimulator") & " " & pdxName())
+
+task device, "build project":
+    nimble "clean"
+    nimble "-d:playdate", "build"
+    make "device"
+
+task all, "build all":
+    nimble "clean"
+    nimble "csim"
+    make "simulator"
+    exec "rm -fR " & nimcacheDir()
+    nimble "cdevice"
+    make "device"
+    exec(sdkPath() & "/bin/pdc Source " & pdxName())
+
+task clean, "clean project":
+    exec "rm -fR " & nimcacheDir()
+    make "clean"
+
+task setup, "Initializes the build structure":
+    ## Creates a default source directory if it doesn't already exist
+
+    # Calling `sdkPath` will ensure the SDK environment variable is saved
+    # to the config path
+    discard sdkPath()
+
+    if not dirExists("Source"):
+        mkDir "Source"
+
+    if not fileExists("Source/pdxinfo"):
+        writeFile(
+            "Source/pdxinfo",
+            [
+                "name=" & projectName(),
+                "author=" & author,
+                "description=" & description,
+                "bundleId=com." & author & "." & projectName()
```

Comment:
May bring issues when the author name or the project name contain spaces.
Leaving this comment here just as a reminder to fix this later.